src/libm-tester/iutsimd.c

```diff
@@ -190,1364 +190,203 @@ uint32_t f2u(float d) {
-  double s[VECTLENDP];
-  int t[VECTLENDP*2];
-  int i;
-  for(i=0;i<VECTLENDP;i++) {
-    s[i] = rand()/(double)RAND_MAX*20000-10000;
-  }
-  int idx = rand() & (VECTLENDP-1);
-  s[idx] = d;
-
-  vdouble a = vloadu_vd_p(s);
-  vint vi = xilogb(a);
-  vstoreu_v_p_vi(t, vi);
-
-  return t[idx];
-}
-#endif
-
 //
 
-#ifdef ENABLE_SP
-
-float xxsinf(float d) {
-  float s[VECTLENSP];
-  int i;
-  for(i=0;i<VECTLENSP;i++) {
-    s[i] = rand()/(float)RAND_MAX*20000-10000;
+#define func_d_d(funcStr, funcName) {		\
+    if (startsWith(buf, funcStr " ")) {		\
+      uint64_t u;					\
+      sscanf(buf, funcStr " %" PRIx64, &u);		\
+      double s[VECTLENDP];				\
+      int i;						\
+      for(i=0;i<VECTLENDP;i++) {			\
+	s[i] = rand()/(double)RAND_MAX*20000-10000;	\
+      }							\
+      int idx = rand() & (VECTLENDP-1);			\
+      s[idx] = u2d(u);					\
+      vdouble a = vloadu_vd_p(s);			\
+      a = funcName(a);					\
+      vstoreu_v_p_vd(s, a);				\
+      u = d2u(s[idx]);					\
+      printf("%" PRIx64 "\n", u);			\
+      fflush(stdout);					\
+      continue;						\
+    }							\
+  }
+
+#define func_d2_d(funcStr, funcName) {		\
+    if (startsWith(buf, funcStr " ")) {		\
+      uint64_t u;				\
+      sscanf(buf, funcStr " %" PRIx64, &u);	\
+      double s[VECTLENDP], t[VECTLENDP];			\
+      int i;							\
+      for(i=0;i<VECTLENDP;i++) {				\
+	s[i] = rand()/(double)RAND_MAX*20000-10000;		\
+	t[i] = rand()/(double)RAND_MAX*20000-10000;		\
+      }								\
+      int idx = rand() & (VECTLENDP-1);				\
+      s[idx] = u2d(u);						\
+      vdouble2 v;						\
+      vdouble a = vloadu_vd_p(s);				\
+      v = funcName(a);						\
+      vstoreu_v_p_vd(s, v.x);					\
+      vstoreu_v_p_vd(t, v.y);					\
+      Sleef_double2 d2;						\
+      d2.x = s[idx];						\
+      d2.y = t[idx];						\
+      printf("%" PRIx64 " %" PRIx64 "\n", d2u(d2.x), d2u(d2.y));	\
+      fflush(stdout);					\
+      continue;						\
+    }							\
+  }
+
+#define func_d_d_d(funcStr, funcName) {	\
+    if (startsWith(buf, funcStr " ")) {	\
+      uint64_t u, v;					\
+      sscanf(buf, funcStr " %" PRIx64 " %" PRIx64, &u, &v);	\
+      double s[VECTLENDP], t[VECTLENDP];			\
+      int i;							\
+      for(i=0;i<VECTLENDP;i++) {				\
+	s[i] = rand()/(double)RAND_MAX*20000-10000;		\
+	t[i] = rand()/(double)RAND_MAX*20000-10000;		\
+      }								\
+      int idx = rand() & (VECTLENDP-1);				\
+      s[idx] = u2d(u);						\
+      t[idx] = u2d(v);						\
+      vdouble a, b;						\
+      a = vloadu_vd_p(s);					\
+      b = vloadu_vd_p(t);					\
+      a = funcName(a, b);						\
+      vstoreu_v_p_vd(s, a);					\
+      u = d2u(s[idx]);						\
+      printf("%" PRIx64 "\n", u);				\
+      fflush(stdout);						\
+      continue;							\
+    }								\
+  }
+
+#define func_d_d_i(funcStr, funcName) {	\
+    if (startsWith(buf, funcStr " ")) {	\
+      uint64_t u, v;					\
+      sscanf(buf, funcStr " %" PRIx64 " %" PRIx64, &u, &v);		\
+      union {								\
```

Comment:
This looks a simple mistake. I will fix this.
Could you tell me a little bit about your "downstream source?"
I can add support for that if possible.